### PR TITLE
APM setup in Agent section 2

### PR DESCRIPTION
**Pull request by l0k0ms**
This is a new PR for: https://github.com/DataDog/documentation/pull/3498

Conflict were too big

#### APM related

* [x] Moves APM Agent setup section in the agent section
* [x] Creates docker sub-section  
* [x] Creates a docker APM sub-sub-section
* [x] Creates a Kubernetes APM sub-sub-section
* [x] Updates K8s default manifest to take into account APM setup
* [x] Creates a Language sub-section in the APM section that list all language instrumentation setup.

#### Side effect

* [x] Moves agent network page to FAQs in order to limit the complexity of the Agent nav 
* [x] Moves custom python page to FAQs in order to limit the complexity of the Agent nav 
* [x] Updates all links 
* [x] Updates all images

### Motivation

After having discussed with APM PMs it has been decided that every part of the APM setup related to the agent should live in the Agent section of the documentation in order to enhance clarity and reduce complexity. 

### Preview link

* [Agent Page](https://docs-staging.datadoghq.com/gus/agent-apm-2/agent/)
* [New Agent APM section](https://docs-staging.datadoghq.com/gus/agent-apm-2/agent/apm/?tab=agent630)
* [K8s APM](https://docs-staging.datadoghq.com/gus/agent-apm-2/agent/kubernetes/apm)
* [New docker section](https://docs-staging.datadoghq.com/gus/agent-apm-2/agent/docker/)
* [Docker APM page](https://docs-staging.datadoghq.com/gus/agent-apm-2/agent/docker/apm/?tab=java)
* [New language section](https://docs-staging.datadoghq.com/gus/agent-apm-2/tracing/languages/)

### CODE REVIEW

File: content/agent/kubernetes/daemonset_setup.md [hunk 1]
```diff
@@ -238,29 +260,37 @@ Learn more about this in [the Docker log collection documentation][9].
...
+        spec:
+          containers:
+          - name: <CONTAINER_NAME>
+            image: <CONTAINER_IMAGE>/<TAG>
+            env:
+              - name: DD_AGENT_HOST
+                valueFrom:
+                  fieldRef:
+                    fieldPath: status.hostIP
+              - name: DD_TRACE_AGENT_PORT
```
Comment (palazzem) [inline content/agent/kubernetes/daemonset_setup.md]: We can remove this, it's `8126` by default and if they change this, it's advanced usage.

File: content/tracing/advanced_usage/_index.md [hunk 2]
```diff
@@ -279,6 +279,88 @@ if (($span = GlobalTracer::get()->getActiveSpan()) !== null){
...
 {{< /tabs >}}
 
+## Change Agent Hostname
+
+Configure your application level tracers to submit traces to a custom Agent hostname. See the examples below for each supported language:
+
+{{< tabs >}}
+{{% tab "Java" %}}
+
+The Java Tracing Module automatically looks for and initializes with the ENV variables DD_AGENT_HOST and DD_TRACE_AGENT_PORT:
```
Comment (palazzem) [inline content/tracing/advanced_usage/_index.md]: This must be replicated for all languages:

The <LANGUAGE> Tracing Module automatically looks for and initializes with the ENV variables `DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT`

File: content/tracing/advanced_usage/_index.md [hunk 3]
```diff
@@ -279,6 +279,88 @@ if (($span = GlobalTracer::get()->getActiveSpan()) !== null){
...
+java -javaagent:/path/to/the/dd-java-agent.jar \
+     -Ddd.agent.host=$DD_AGENT_HOST \
+     -Ddd.agent.port=$DD_TRACE_AGENT_PORT \
+     -jar /your/app.jar
+```
+
+{{% /tab %}}
+{{% tab "Python" %}}
+
+```python
```
Comment (palazzem) [inline content/tracing/advanced_usage/_index.md]: After adding the previous comment, you can say: "You can also use the Tracer configuration API" (or similar)

File: content/tracing/advanced_usage/_index.md [hunk 4]
```diff
@@ -279,6 +279,88 @@ if (($span = GlobalTracer::get()->getActiveSpan()) !== null){
 {{% /tab %}}
 {{< /tabs >}}
 
+## Change Agent Hostname
```
Comment (palazzem) [inline content/tracing/advanced_usage/_index.md]: Not sure, but in the preview I don't see this section in the navigation menu (on the left)

File: content/tracing/advanced_usage/_index.md (hunk 2 above)
Comment (palazzem) [reply content/tracing/advanced_usage/_index.md]: (also backtick are missing in DD_AGENT_HOST and DD_TRACE_AGENT_PORT)

File: content/agent/kubernetes/daemonset_setup.md [hunk 5]
```diff
@@ -238,29 +260,37 @@ Learn more about this in [the Docker log collection documentation][9].
 
 ### Trace Collection
 
-To enable [Trace collection][10] with your DaemonSet:
+To enable trace collection with your DaemonSet:
```
Comment (palazzem) [inline content/agent/kubernetes/daemonset_setup.md]: This section is super important. It must stay here but I'd say to add a link for all languages? So:
* keep the section here
* in all languages (such as https://docs-staging.datadoghq.com/gus/agent-apm-2/tracing/languages/java/) change `see additional documentation for tracing Docker applications` in `see additional documentation for tracing Docker applications and Kubernetes` with a link to this section

In this way a random team that has the Agent already deployed can just open the language section, see Kubernetes needs to rebuild the container, install the tracer, add the following config, and they're done.

Makes sense?

File: content/agent/kubernetes/daemonset_setup.md [hunk 6]
```diff
@@ -140,6 +159,9 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: status.hostIP
+          # Trace Collection (APM) - comment this section to disable APM
```
Comment (hkaj) [inline content/agent/kubernetes/daemonset_setup.md]: I'm repeating myself here, but it makes it really confusing to the user imo. We have one section here that they must `comment to disable APM`, and one line above that they should `uncomment to enable APM`. So what's the state by default? Hard to say. It's actually "enabled, harmless, but not actually usable` :/

Comment (l0k0ms) [reply content/agent/kubernetes/daemonset_setup.md]: I'm tracking this feedback on our board, we will address it in another PR

Comment (palazzem) [reply content/agent/kubernetes/daemonset_setup.md]: The process is not expensive if it does nothing (it's stuck listening to a channel so the Go runtime will optimize that) so as we discussed in our RFC, it's not a real issue. Users should enable APM easily so the "uncomment" is good. Maybe we can just remove this comment and that's all.

The end result is: "the Trace Agent doesn't collect traces, if you uncomment we'll start receiving your traces"

**Pull request merged**